github/src/utils/gitHubUrlParse.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface ParsedGitHubUrl {
+    urlReference?: string;
+    ownerOrOrganization?: string;
+    repositoryName?: string;
+}
+
+export function gitHubUrlParse(url?: string): ParsedGitHubUrl {
```

Comment:
Is there any reason to make `url` an optional argument? I feel like there's no point in calling it if you don't have a url to begin with.

Then `ParsedGitHubUrl` could have every property be required. It feels like it'd be annoying to have to type check that any time that you wanted to use `gitHubUrlParse`